src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/AddressServiceImpl.java

```diff
@@ -1,15 +1,95 @@
 package uk.gov.ons.ctp.integration.contactcentresvc.service.impl;
 
-import com.godaddy.logging.Logger;
-import com.godaddy.logging.LoggerFactory;
+import java.util.ArrayList;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+import uk.gov.ons.ctp.integration.contactcentresvc.client.AddressServiceClientServiceImpl;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.AddressDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.AddressQueryRequestDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.AddressQueryResponseDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.representation.PostcodeQueryRequestDTO;
 import uk.gov.ons.ctp.integration.contactcentresvc.service.AddressService;
+import uk.gov.ons.ctp.integration.contactcentresvc.service.addressindex.model.AddressIndexAddressDTO;
+import uk.gov.ons.ctp.integration.contactcentresvc.service.addressindex.model.AddressIndexSearchResultsDTO;
 
 /**
  * A ContactCentreDataService implementation which encapsulates all business logic for getting
  * Addresses
  */
 @Service
+@Validated()
 public class AddressServiceImpl implements AddressService {
-  private static final Logger log = LoggerFactory.getLogger(AddressServiceImpl.class);
+
+  @Autowired private AddressServiceClientServiceImpl addressServiceClient;
+
+  @Override
+  public AddressQueryResponseDTO addressQuery(AddressQueryRequestDTO addressQueryRequest) {
+    // Delegate the query to Address Index
+    AddressIndexSearchResultsDTO addressIndexResponse =
+        addressServiceClient.addressQuery(addressQueryRequest);
+
+    // Summarise the returned addresses
+    return convertAddressIndexResultsToSummarisedAdresses(addressIndexResponse);
+  }
+
+  @Override
+  public AddressQueryResponseDTO postcodeQuery(PostcodeQueryRequestDTO postcodeQueryRequest) {
+    // Delegate the query to Address Index
+    AddressIndexSearchResultsDTO addressIndexResponse =
+        addressServiceClient.postcodeQuery(postcodeQueryRequest);
+
+    // Summarise the returned addresses
+    return convertAddressIndexResultsToSummarisedAdresses(addressIndexResponse);
+  }
+
+  private AddressQueryResponseDTO convertAddressIndexResultsToSummarisedAdresses(
+      AddressIndexSearchResultsDTO addressIndexResponse) {
+    ArrayList<AddressDTO> summarisedAddresses = new ArrayList<>();
+    for (AddressIndexAddressDTO fullAddress : addressIndexResponse.getResponse().getAddresses()) {
```

Comment:
will getResponse always be != null ?

Reply:
Yes. 
AI works on returning an empty result structure instead of the 404 that I had initially expected.

Here is an example call to the AI that doesn't find anything:
```
$ curl -s "http://addressindex-api-dev.apps.devtest.onsclofo.uk/addresses?input=ZZZZZZZZZZZZ" | jq
{
  "apiVersion": "v_c875fa34fdfa1cf308505b8b6524dbe5e4824595",
  "dataVersion": "39",
  "response": {
    "tokens": {
      "TownName": "ZZZZZZZZZZZZ"
    },
    "addresses": [],
    "filter": "",
...
```